### ISSUE TITLE
TypeScript error when trying to use config in extends

### ISSUE THREAD

**Issue by TrevorBurnham** [un-ts/eslint-plugin-import-x#439]
Consider this `eslint.config.mjs`:

```mjs
// @ts-check
import { defineConfig } from "eslint/config";
import eslint from "@eslint/js";
import importX from "eslint-plugin-import-x";

export default defineConfig({
  ignores: ["build/"],
  extends: [
    eslint.configs.recommended,
    importX.configs.recommended // TypeScript error
  ],
});
```

tsc complains:

```
mport-x/no-duplicates': "warn"; }; parserOpt...' is not assignable to type 'ExtendsElement'.
  Type '{ plugins: ["import-x"]; rules: { 'import-x/no-unresolved': "error"; 'import-x/named': "error"; 'import-x/namespace': "error"; 'import-x/default': "error"; 'import-x/export': "error"; 'import-x/no-named-as-default': "warn"; 'import-x/no-named-as-default-member': "warn"; 'import-x/no-duplicates': "warn"; }; parserOpt...' is not assignable to type 'ConfigObject<RulesConfig>'.
    Types of property 'plugins' are incompatible.
      Type '["import-x"]' is not assignable to type 'Record<string, Plugin>'.
        Index signature for type 'string' is missing in type '["import-x"]'.

8   extends: [eslint.configs.recommended, importX.configs.recommended],
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in eslint.config.mjs:8
```

I haven't seen this issue with any other ESLint library that exports flat configs.

Here's a reproduction: https://codesandbox.io/p/devbox/objective-ganguly-h4h4v7

**Comment by TrevorBurnham** [comment]
Aha! I was able to reproduce my original issue with `flatConfigs`; it's just that the issue only occurs if you install a recent version of typescript-eslint, because of how it globally overrides ESLint's `defineConfig` types. The issue was already reported over there: https://github.com/typescript-eslint/typescript-eslint/issues/11543

Reopening as I think it's worth considering updating the library's types to deal with this. I'm sure a large number of eslint-plugin-import-x's users also use typescript-eslint.

I've updated the PR description.

**Comment by zagoorland** [comment]
I have similar typescript error when im using plugin like this:


```ts
export default defineConfig([
  globalIgnores(['dist']),
  {
    files: ['**/*.{ts,tsx}'],
    plugins: {
      'import-x': importX,
    },
...
```

```ts
Type '{ meta: { name: string; version: string; }; rules: { 'no-unresolved': RuleModule<MessageId, [(Options | undefined)?], ImportXPluginDocs, RuleListener>; ... 45 more ...; 'imports-first': RuleModule<...>; }; cjsRequire: CjsRequire; importXResolverCompat: (resolver: LegacyResolver | NewResolver, resolverOptions?: unkno...' is not assignable to type 'Plugin'.
  Types of property 'configs' are incompatible.
    Type '{ recommended: { plugins: ["import-x"]; rules: { 'import-x/no-unresolved': "error"; 'import-x/named': "error"; 'import-x/namespace': "error"; 'import-x/default': "error"; 'import-x/export': "error"; 'import-x/no-named-as-default': "warn"; 'import-x/no-named-as-default-member': "warn"; 'import-x/no-duplicates': "warn...' is not assignable to type 'Record<string, ConfigObject<RulesConfig> | LegacyConfigObject<RulesConfig, RulesConfig> | ConfigObject<RulesConfig>[]>'.
      Property ''stage-0'' is incompatible with index signature.
        Type 'PluginConfig' is not assignable to type 'ConfigObject<RulesConfig> | LegacyConfigObject<RulesConfig, RulesConfig> | ConfigObject<RulesConfig>[]'.
          Type 'PluginConfig' is not assignable to type 'LegacyConfigObject<RulesConfig, RulesConfig>'.
            Types of property 'overrides' are incompatible.
              Type 'ConfigOverride[] | undefined' is not assignable to type 'ConfigOverride<RulesConfig>[] | undefined'.
                Type 'ConfigOverride[]' is not assignable to type 'ConfigOverride<RulesConfig>[]'.
                  Type 'ConfigOverride' is not assignable to type 'ConfigOverride<RulesConfig>'.
                    Types of property 'parser' are incompatible.
                      Type 'string | null | undefined' is not assignable to type 'string | undefined'.
                        Type 'null' is not assignable to type 'string | undefined'.
```

**Comment by SukkaW** [comment]
Semi duplicated as https://github.com/un-ts/eslint-plugin-import-x/issues/421

Let's track this in that issue instead.
